lib/p2p/Peer.ts

```diff
@@ -131,7 +131,12 @@ class Peer extends EventEmitter {
 
     await this.initConnection();
     this.initStall();
-    await this.initHello(handshakeData);
+    this.sendHello(handshakeData);
+
+    if (!this.handshakeState) {
+      // we must wait to receive handshake data before opening the connection
+      await this.wait(PacketType.HELLO, Peer.RESPONSE_TIMEOUT);
+    }
```

Comment:
I would still keep it inside `initHello`, to make `open` a bit cleaner. Not a must